homeassistant/components/binary_sensor/bayesian.py

```diff
@@ -0,0 +1,225 @@
+"""
+Use Bayesian Inference to trigger a binary sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.bayesian/
+"""
+import asyncio
+import logging
+from collections import OrderedDict
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.binary_sensor import (BinarySensorDevice,
+                                                    PLATFORM_SCHEMA)
+from homeassistant.const import (CONF_ABOVE, CONF_BELOW, CONF_DEVICE_CLASS,
+                                 CONF_ENTITY_ID, CONF_NAME, CONF_PLATFORM,
+                                 CONF_STATE, STATE_UNKNOWN)
+from homeassistant.core import callback
+from homeassistant.helpers import condition
+from homeassistant.helpers.event import async_track_state_change
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_OBSERVATIONS = 'observations'
+CONF_PRIOR = 'prior'
+CONF_PROBABILITY_THRESHOLD = 'probability_threshold'
+CONF_P_GIVEN_F = 'prob_given_false'
+CONF_P_GIVEN_T = 'prob_given_true'
+CONF_TO_STATE = 'to_state'
+
+DEFAULT_NAME = 'BayesianBinary'
+
+NUMERIC_STATE_SCHEMA = vol.Schema({
+    CONF_PLATFORM:
+    'numeric_state',
+    CONF_ENTITY_ID:
+    cv.string,
+    vol.Optional(CONF_ABOVE):
+    vol.Coerce(float),
+    vol.Optional(CONF_BELOW):
+    vol.Coerce(float),
+    CONF_P_GIVEN_T:
+    vol.Coerce(float),
+    vol.Optional(CONF_P_GIVEN_F):
+    vol.Coerce(float)
+})
```

Comment:
You can set all the nodes to be required in the schema except the ones marked as optional with the `required` keyword argument.
```py
vol.Schema({...}, required=True)
```